recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/python3
+from optparse import OptionParser
+import sys
+
+force_update=False
+
+def sha256_file(file,length):
+    """
+    Return sha256 of the file
+    """
+    import mmap
+    import hashlib
+    method=hashlib.sha256()
+    try:
+        with mmap.mmap(file.fileno(),length,access=mmap.ACCESS_READ) as mm:
+            chunk = mm.read(length)
+            method.update(chunk)
+    except:
+         pass
+    return method.hexdigest()
+
+class FirmwareUpdate(object):
+    def __init__(self,force,erase):
+        self.force_update=force
+        self.erase_env=erase
+
+    def sha256_check(self,file):
+        """
+            Firmware Sha256 Check
+        """
+        check_sha256_file=file+'.sha256'
+        
+        try:
+            with open(check_sha256_file, 'r') as f:
+                check_sha256=f.read()
+        except:
+            raise FirmwareUpdateException("sha256 file not exists")
+
+        try:
+            with open(file, 'rb+') as f:
+                hash_code=sha256_file(f,0)
+        except:
+            raise FirmwareUpdateException("Firmware not Exists")
+        
+        if check_sha256 != hash_code:
+            if( self.force_update != True):
+                raise FirmwareUpdateException("File Checksum is not Correct")
```

Comment:
I suspect those exceptions will give rather verbose error reportings when they occur. I would recommend using plain error messages and sys.exit(1) instead.